examples/server-side-rendering/src/app/app.component.ts

```diff
@@ -2,10 +2,27 @@ import { Component, Injector, Inject, PLATFORM_ID } from '@angular/core';
 import { isPlatformServer } from '@angular/common';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { TransferState, makeStateKey } from '@angular/platform-browser';
-import {
-  createSSRSearchClient,
-  parseServerRequest,
-} from 'angular-instantsearch';
+import { history } from 'instantsearch.js/es/lib/routers';
+import { simple } from 'instantsearch.js/es/lib/stateMappings';
+import { StateMapping } from 'instantsearch.js/es/types';
+import { createSSRSearchClient } from 'angular-instantsearch';
+import * as qs from 'qs';
+
+function parseServerRequest(
+  req: { url: string } | undefined,
+  stateMapping: StateMapping
+) {
+  if (!req || !req.url) {
+    return undefined;
+  }
+
+  // Transform the URL into RouteState, this is likely the same
+  // implementation as router.read(), but without reading from the environment.
+  const routeState = qs.parse(req.url.split('?')[1], { arrayLimit: 99 });
+
+  // Transform from RouteState into UiState using the stateMapping
+  return stateMapping.routeToState(routeState);
```

Comment:
Why do we still need `stateMapping`? This shouldn't be necessary.

Reply:
initialUiState is UiState, not routeState, the only reason this works is because the simple state mapping doesn't really map, but I don't want to give a solution which fails if you have somewhat custom routing needs

Reply:
Again, we shouldn't use IS.js internals in user code. We'd better off inlining the logic here (or worst case import it from Angular InstantSearch).

Reply:
I'm not following, this is API, not internals

Reply:
I got confused reading this component, I thought the last part was in the library.

My point is that we don't need to use the routing API and these indirections. This would be easier to understand for users:

```ts
export class AppComponent {
  public instantsearchConfig: {};

  constructor(
    private httpClient: HttpClient,
    private transferState: TransferState,
    private injector: Injector,
    @Inject(PLATFORM_ID) private platformId: Object
  ) {
    const req = isPlatformServer(this.platformId)
      ? this.injector.get('request')
      : { url: '' };

    const queryParams = req.url.slice(req.url.lastIndexOf('?') + 1);
    const initialUiState = qs.parse(queryParams, { arrayLimit: 99 });

    this.instantsearchConfig = {
      indexName: 'instant_search',
      routing: true,
      initialUiState,
      searchClient: createSSRSearchClient({
        appId: 'latency',
        apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
        makeStateKey,
        HttpHeaders,
        transferState: this.transferState,
        httpClient: this.httpClient,
      }),
    };
  }
}
```

Reply:
I don't understand why you wouldn't use the routing API, if it's the same code as it

Reply:
Because it's not needed, that's one more API for users to understand, more indirection, more code, more dependencies.

Reply:
your code only works for an identity stateMapping

Reply:
Which is the point of this example, right?

That's the way [we document it in other flavors](https://github.com/algolia/react-instantsearch/blob/1cb1b3313c066a8eff5cc3d1f30ff335c8d81adc/examples/next/pages/index.js#L19-L20).

Reply:
in react we don't have routing though, so I don't think it's a good comparison

Reply:
Ok, let's use `stateMapping` then 👍

Reply:
The thing is stateMapping is not enough for most routings. (you can't change the path for example)
Our SEO examples we created (because it's the usual application of SSR) need [more than a simple stateMapping](https://github.com/algolia/doc-code-samples/blob/master/InstantSearch.js/routing-seo-friendly/src/search-routing.js).